micro-rdk/src/common/grpc.rs

```diff
@@ -1435,35 +1439,99 @@ impl<'a> GrpcServerInner<'a> {
         let rr = robot::v1::ResourceNamesResponse { resources: rr };
         self.encode_message(rr)
     }
+
+    fn signaling_service_optional_webrtc_config(
+        &mut self,
+        message: &[u8],
+    ) -> Result<Bytes, ServerError> {
+        let req = proto::rpc::webrtc::v1::OptionalWebRtcConfigRequest::decode(message)
+            .map_err(|_| ServerError::from(GrpcError::RpcInvalidArgument))?;
+        let result = match self.signaling_server {
+            Some(ref ss) => ss.optional_webrtc_config(req),
+            None => Err(ServerError::from(GrpcError::RpcUnimplemented)),
+        }?;
+        self.encode_message(result)
+    }
+
+    fn signaling_service_call(
+        self,
+        message: &[u8],
+    ) -> Pin<Box<dyn futures_lite::Stream<Item = Result<Bytes, ServerError>> + Sync + Send>> {
+        let (sender, receiver) = async_channel::bounded::<Result<CallResponse, ServerError>>(1);
+
+        match self.signaling_server {
+            Some(ss) => {
+                let cr = proto::rpc::webrtc::v1::CallRequest::decode(message);
+                let ss = ss.clone();
+                ss.executor
+                    .clone()
+                    .spawn(async move {
+                        match cr {
+                            Ok(cr) => match ss.call(cr, sender.clone()).await {
```

Comment:
I think this pattern of having
```
async fn(request, sender) -> Result<(), Error>
```

Is going to be somewhat common. You want to be able to write the function normally with `?` and such to check preconditions, but then once you are into the streaming part you are sending results/errors to the channel, rather than returning them.

I'd like to have a combinator that wraps a function of that shape and feeds any `Err` result back into the channel.